### PR TITLE
iOS Auth module

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [ ] Are tests passing locally?
* [ ] Are the files formatted correctly?
* [ ] Did you add unit tests if this modifies the UWP implementation?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Related PRs or issues

[AB#60472](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/60472)

### CODE REVIEW

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs [hunk 1]
```diff
@@ -10,8 +11,17 @@ public partial class Auth : AppCenterService
     {
         private static Task<SignInResult> PlatformSignInAsync()
         {
-            MSIdentity.SignIn(null);
-            return null;
+            var promise = new TaskCompletionSource<SignInResult>();
```
Comment (MatkovIvan) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs]: Let's don't use "promise" in naming here

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs [hunk 2]
```diff
@@ -10,8 +11,17 @@ public partial class Auth : AppCenterService
...
+
+            MSIdentity.SignIn((userInformation, error) =>
+            {
+                SignInResult result = new SignInResult();
+                result.UserInformation = new UserInformation();
+                result.UserInformation.AccountId = userInformation.AccountId;
+                result.Exception = new Exception(error.LocalizedDescription);
```
Comment (MatkovIvan) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs]: Null checks required

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (guperrot) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Properties/AssemblyInfo.cs]: Align AssemblyInfo version fields with existing projects.

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+﻿using System;
```
Comment (MatkovIvan) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs]: Add copyright to new files

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs (hunk 1 above)
Comment (Zongrenfeng) [reply SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs]: changed to taskCompletionSource.

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs (hunk 4 above)
Comment (Zongrenfeng) [reply SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs]: Added

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+using ObjCRuntime;
+using Foundation;
```
Comment (jwallra) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs]: Order `using`s

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    // @interface MSIdentity : MSService
+    [BaseType(typeof(NSObject))]
+    interface MSIdentity
+    {
+        // +(void)setEnabled:(BOOL)isEnabled;
+        [Static]
+        [Export("setEnabled:")]
+        void SetEnabled(bool isEnabled);
+
+        // +(BOOL)isEnabled;
```
Comment (jwallra) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs]:
```suggestion
        // + (BOOL)isEnabled;
```

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+using Foundation;
+using UIKit;
+
+namespace Microsoft.AppCenter.Auth.iOS.Bindings
+{
+    // @interface MSIdentity : MSService
+    [BaseType(typeof(NSObject))]
+    interface MSIdentity
+    {
+        // +(void)setEnabled:(BOOL)isEnabled;
```
Comment (jwallra) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs]:
```suggestion
        // + (void)setEnabled:(BOOL)isEnabled;
```

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        [Static]
+        [Export("setEnabled:")]
+        void SetEnabled(bool isEnabled);
+
+        // +(BOOL)isEnabled;
+        [Static]
+        [Export("isEnabled")]
+        bool IsEnabled();
+
+        // + (void) signInWithCompletionHandler:(MSSignInCompletionHandler _Nullable) completionHandler;
```
Comment (jwallra) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs]:
```suggestion
        // + (void)signInWithCompletionHandler:(MSSignInCompletionHandler _Nullable) completionHandler;
```

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+        [Static]
+        [Export("signInWithCompletionHandler:")]
+        void SignIn(MSSignInCompletionHandler completionHandler);
+
+        // + (void) signOut;
```
Comment (jwallra) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs]:
```suggestion
        // + (void)signOut;
```

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+        [Static]
+        [Export("signOut")]
+        void SignOut();
+
+        // + (void) setConfigUrl:(NSString*) configUrl;
```
Comment (jwallra) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/ApiDefinition.cs]:
```suggestion
        // + (void)setConfigUrl:(NSString*) configUrl;
```

File: scripts/configuration/ac-build-config.xml [hunk 11]
```diff
@@ -2,7 +2,7 @@
 <config>
     <sdkInfo>
         <sdkVersion>2.0.0-SNAPSHOT</sdkVersion>
-        <iosVersion>1.13.2</iosVersion>
+        <iosVersion>2.0.0-189+1f5f7725098b3e661f17369ab2da793d82cae674</iosVersion>
```
Comment (jwallra) [inline scripts/configuration/ac-build-config.xml]: Is this a SHA ID appended to the version?  It just seems odd that it changed like this between 1.x and 2.x

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+            var taskCompletionSource = new TaskCompletionSource<UserInformation>();
+            MSIdentity.SignIn((userInformation, error) =>
+            {
+                if (error != null)
+                {
+                    throw new NSErrorException(error);
+                }
+                taskCompletionSource.TrySetResult(new UserInformation
+                {
+                    AccountId = userInformation?.AccountId
```
Comment (dhei) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs]: Is `?` really needed here? Native iOS sdk should return either an non-nil `error` OR a non-nil `userInformation`, but not both.

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/Structs.cs [hunk 13]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.AppCenter.Auth.iOS.Bindings
+{
+}
+
```
Comment (dhei) [inline SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS.Bindings/Structs.cs]:
```suggestion
```
Remove empty line.

File: SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs (hunk 12 above)
Comment (guperrot) [reply SDK/AppCenterAuth/Microsoft.AppCenter.Auth.iOS/Auth.cs]: I agree.

File: scripts/configuration/ac-build-config.xml (hunk 11 above)
Comment (guperrot) [reply scripts/configuration/ac-build-config.xml]: That's normal on internal repo.

**Pull request merged**